src/leafdoc.js

```diff
@@ -1,837 +0,0 @@
-	this._miniclasses = {};
-
-	this._AKAs = {};
-
-	// 🍂section
-	// 🍂aka Leafdoc options
-	if (options) {
-		// 🍂option templateDir: String = 'templates/basic'
-		// Defines which subdirectory (relative to the directory the curent JS
-		// script is running) holds the handlebars template files for building up the HTML.
-		if (options.templateDir) {
-			setTemplateDir(options.templateDir);
-		}
-
-		// 🍂option showInheritancesWhenEmpty: Boolean = false
-		// When `true`, child classes/namespaces will display documentables from ancestors, even if the child class doesn't have any of such documentables.
-		// e.g. display inherited events even if the child doesn't define any new events.
-		this.showInheritancesWhenEmpty = options.showInheritancesWhenEmpty || false;
-
-		// 🍂option leadingCharacter: String = '🍂'
-		// Overrides the Leaf symbol as the leading character for documentation lines.
-		// See also [`setLeadingCharacter`](#leafdoc-setleadingcharacter).
-		if (options.leadingCharacter) {
-			this.setLeadingCharacter(options.leadingCharacter);
-		}
-
-		// 🍂option customDocumentables: Map = {}
-		// A key-value map. Each pair will be passed to [`registerDocumentable`](#leafdoc-registerdocumentable).
-		if (options.customDocumentables) {
-			for (var i in options.customDocumentables) {
-				this.registerDocumentable(i, options.customDocumentables[i]);
-			}
-		}
-
-		// 🍂option verbose: Boolean = false
-		// Set to `true` to display more information as files are being read.
-		if (options.verbose) {
-			this._verbose = options.verbose;
-		}
-	}
-}
-
-/*
-🍂constructor Leafdoc(options: Leafdoc options); Constructor for a new Leafdoc parser
-
-🍂example
-
-Output Leafdoc's own documentation to the console with:
-
-```
-var LeafDoc = require('./src/leafdoc.js');
-var doc = new LeafDoc();
-	doc.addFile('src/leafdoc.js');
-
-console.log( doc.outputStr() );
-```
- */
-
-
-// 🍂method registerDocumentable (name: String, label?: String, inheritable?: Boolean): this
-// Registers a new documentable type, beyond the preset ones (function,
-// property, etc). New documentable should also not be an already used
-// keyword (class, namespace, inherits, etc).
-// When registering new documentables, make sure that there is an appropriate
-// template file for it.
-// Set `label` to the text for the sections in the generated docs.
-// `inheritable` parameter determines documentable can be inherited via inherits keyword in a subclass.
-Leafdoc.prototype.registerDocumentable = function (name, label, inheritable) {
-
-	this._knownDocumentables.push(name, label);
-
-	if (label) {
-		this._documentableLabels[name] = label;
-	}
-
-	if (inheritable) {
-		this._inheritableDocumentables.push(name);
-	}
-
-	return this;
-};
-
-// 🍂method getTemplateEngine(): Handlebars
-// Returns handlebars template engine used to render templates.
-// You can use it for override helpers or register new.
-Leafdoc.prototype.getTemplateEngine = function () {
-	return template.engine;
```

Comment:
This was a bug, `template` was undefined, which TypeScript pointed out after conversion. Note the import above:

```js
import {getTemplate, setTemplateDir, setAKAs} from './template';
```